tests/context/context_constructors.cpp

```diff
@@ -86,8 +94,23 @@ class TEST_NAME : public util::test_base {
         auto deviceList = platform.get_devices();
         sycl::context context(deviceList);
 
-        if (context.is_host() != selector.is_host()) {
-          FAIL(log, "context was not constructed correctly (is_host)");
+        if (context.get_devices().size() != deviceList.size()) {
+          FAIL(log, "context was not constructed correctly (get_devices size)");
+        }
+
+        for (auto &device : context.get_devices()) {
+          if (std::find(deviceList.begin(), deviceList.end(), device) ==
+              deviceList.end()) {
+            FAIL(log,
+                 "context was not constructed correctly (device not in passed "
+                 "device list)");
+          }
+
+          if (std::count(context.get_devices().begin(),
```

Comment:
You could just count once and gives 2 different error messages according to the answer.

Reply:
With just this call you can also replace the previous `std::find`. If it is `>= 2` there are too many devices, if `== 0` the device is not found. Only 1 O(n) algorithm instead of 2.

Reply:
I see! Technically it would change the check as one checks for the device in the device-list from the platform and the other counts it in the list of devices reported by the context. However, the duplicate check is somewhat redundant as it will be caught by the other two checks, unless the platform has duplicate devices too which would arguably be platform issue rather than a context issue.